src/pages/api/token/pr/index.ts

```diff
@@ -0,0 +1,173 @@
+import { createNotification } from '@/lib/notificationModels';
+import {
+   createPaymentRequest,
+   getPayentRequestById,
+   markPaymentRequestAsPaid,
+} from '@/lib/paymentRequestModels';
+import { createTokenInDb } from '@/lib/tokenModels';
+import { findUserByPubkeyWithMint } from '@/lib/userModels';
+import {
+   AuthenticatedRequest,
+   Currency,
+   GetPaymentRequestResponse,
+   NotificationType,
+} from '@/types';
+import { computeTxId, initializeWallet } from '@/utils/cashu';
+import { corsMiddleware, runAuthMiddleware, runMiddleware } from '@/utils/middleware';
+import { getBaseURLFromRequest } from '@/utils/url';
+import {
+   PaymentRequestTransportType,
+   PaymentRequestTransport,
+   PaymentRequestPayload,
+   PaymentRequest,
+   getEncodedTokenV4,
+} from '@cashu/cashu-ts';
+import { NextApiResponse, NextApiRequest } from 'next';
+
+type GetPaymentRequestQuery = {
+   amount?: number;
+   reusable?: boolean;
+};
+
+export default async function handler(
+   req: AuthenticatedRequest | NextApiRequest,
+   res: NextApiResponse,
+) {
+   await runMiddleware(req, res, corsMiddleware);
+
+   // Handle OPTIONS request
+   if (req.method === 'OPTIONS') {
+      res.setHeader('Allow', ['GET', 'POST', 'OPTIONS']);
+      return res.status(200).end();
+   }
+
+   if (req.method === 'GET') {
+      await runAuthMiddleware(req as AuthenticatedRequest, res);
+      const pubkey = (req as AuthenticatedRequest).authenticatedPubkey;
+      const { amount, reusable } = req.query as unknown as GetPaymentRequestQuery;
+      if (!pubkey) {
+         return res.status(400).json({ error: 'Missing required parameters' });
+      }
+      if (amount && isNaN(Number(amount))) {
+         return res.status(400).json({ error: 'amount must be a number' });
+      }
+      if (typeof pubkey !== 'string') {
+         return res.status(400).json({ error: 'pubkey must be a string' });
+      }
+
+      const user = await findUserByPubkeyWithMint(pubkey);
+
+      if (!user) {
+         return res.status(404).json({ error: 'User not found' });
+      }
+
+      const mint: string = user.defaultMint.url;
+      const unit = user.defaultUnit as Currency;
+
+      const httpTransport: PaymentRequestTransport = {
+         type: PaymentRequestTransportType.POST,
+         target: `${getBaseURLFromRequest(req)}/api/token/pr`,
+      };
+
+      const { id } = await createPaymentRequest({
```

Comment:
so that when the payment request is paid we can find the details about it.

- Alice request a payment request
- create payment request in db
- Bob pays the payment request
- we lookup the request that got paid and notify Alice she received the payment

Reply:
I see. I forgot that the receiver specifies the http transport which is again our endpoint and then the client is polling